lib/auth/result.rb

```diff
@@ -137,9 +137,18 @@ def to_client_hash
       return result
     end
 
+    suggested_username = UserNameSuggester.suggest(username_suggester_attributes)
+    if email_valid && email.present?
+      if username.present? && User.username_available?(username, email)
+        suggested_username = username
+      elsif staged_user = User.where(staged: true).find_by_email(email)
```

Comment:
I don't think we need this branch... it's already covered by the logic in `User#username_available?`

https://github.com/discourse/discourse/blob/2f5e63fd7f8a1598882a35bd216d8eb59bb73d5a/app/models/user.rb#L282

Reply:
I checked it, I think that we need that fallback in case the username is not provided by the authenticator